src/js/game/LevelMVC/LevelModel.js

```diff
@@ -345,8 +345,17 @@ module.exports = class LevelModel {
     return Position.forward(entity.position, entity.facing);
   }
 
-  getMoveBackwardPosition(entity = this.player) {
-    return Position.forward(entity.position, FacingDirection.opposite(entity.facing));
+  getMoveDirectionPosition(entity, direction) {
+    switch (direction) {
+      case FacingDirection.North:
+        return Position.forward(entity.position, entity.facing);
+      case FacingDirection.South:
+        return Position.forward(entity.position, FacingDirection.opposite(entity.facing));
+      case FacingDirection.East:
+        return Position.forward(entity.position, FacingDirection.right(entity.facing));
+      case FacingDirection.West:
+        return Position.forward(entity.position, FacingDirection.left(entity.facing));
+    }
```

Comment:
why is this assuming so strongly that the entity is always facing north, rather than just using `Position.forward(entity.position, direction)` for all cases?

Reply:
It's not assuming the entity is facing north?

for args: entity is the entity doing the check (this is currently used in placeDirection only I believe), and direction is the direction relative to that entity (so 2 for instance would not be absolute down, it would indicate -behind- whatever direction the entity is facing).

So if direction === 2, our case hits FacingDirection.South, which is honestly the worst part of this, but I tried to use the FacingDirection enum there rather than just numbers so we would have some semblance of what they mean.

In case FacingDireciton.South:
we're going to return position.Forward (the index "in front" of the entity) given the entity position, and the facing direction relative to that entity. 

So for behind that's: FacingDirection.opposite(entity.facing)); meaning the index that is opposite the direction the entity is facing.

Or left, for case West:, right for case East:, and so on.